src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/adjudications/AdjudicationService.kt

```diff
@@ -549,8 +551,41 @@ class AdjudicationService(
   }
 
   fun getHearingResult(hearingId: Long, resultSeq: Int = 1): HearingResult =
-    adjudicationHearingResultRepository.findByIdOrNull(AdjudicationHearingResultId(hearingId, resultSeq))?.toHearingResult()
+    adjudicationHearingResultRepository.findByIdOrNull(AdjudicationHearingResultId(hearingId, resultSeq))
+      ?.toHearingResult()
       ?: throw NotFoundException("Hearing Result not found. Hearing Id: $hearingId, result sequence: $resultSeq")
+
+  fun deleteHearingResult(adjudicationNumber: Long, hearingId: Long) {
+    // allow delete request to fail if adjudication doesn't exist as should never happen
+    adjudicationIncidentPartyRepository.findByAdjudicationNumber(adjudicationNumber)
+      ?: throw NotFoundException("Hearing with id $hearingId delete failed: Adjudication party with adjudication number $adjudicationNumber not found")
```

Comment:
Would it be better to just return in this scenario? Delete is supposed to be idempotent.

Reply:
Oh I see, this is the parent entity. Ignore me 😄